### PR TITLE
Consistently layout connections previews

### PR DESCRIPTION
**Pull request by barmac**
This unifies connection preview features already implemented in several components (Bendpoints, Create, Connect/GlobalConnect) within ConnectionPreview. 

Related https://github.com/bpmn-io/bpmn-js/issues/744

### CODE REVIEW

File: lib/features/bendpoints/BendpointMovePreview.js [hunk 1]
```diff
@@ -23,20 +23,12 @@ var HIGH_PRIORITY = 1100;
 /**
  * A component that implements moving of bendpoints
  */
-export default function BendpointMove(injector, eventBus, canvas, graphicsFactory) {
-
-  // optional components
-  var connectionDocking = injector.get('connectionDocking', false),
-      layouter = injector.get('layouter', false);
+export default function BendpointMove(injector, eventBus, canvas) {
```
Comment (philippfromme) [inline lib/features/bendpoints/BendpointMovePreview.js]: Rename to BendpointMovePreview.

File: lib/features/bendpoints/BendpointSnapping.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import {
 var abs= Math.abs,
     round = Math.round;
 
-var TOLERANCE = 10;
+var TOLERANCE = 5;
```
Comment (philippfromme) [inline lib/features/bendpoints/BendpointSnapping.js]: This is problematic. We shouldn't have to care if grid snapping exists or not.

Comment (barmac) [reply lib/features/bendpoints/BendpointSnapping.js]: I think we should make bendpoint snapping totally optional and then remove it from bpmn-js. What's your opinion on that?

File: lib/features/bendpoints/BendpointMovePreview.js (hunk 1 above)
Comment (barmac) [reply lib/features/bendpoints/BendpointMovePreview.js]:
```suggestion
export default function BendpointMovePreview(injector, eventBus, canvas) {
```

File: lib/features/bendpoints/BendpointMovePreview.js [hunk 3]
```diff
@@ -164,13 +145,14 @@ export default function BendpointMove(injector, eventBus, canvas, graphicsFactor
...
     }
 
-    redrawConnection(e);
+    if (connectionPreview) {
+      connectionPreview.cleanUp(context);
+    }
   });
 }
 
 BendpointMove.$inject = [
```
Comment (barmac) [inline lib/features/bendpoints/BendpointMovePreview.js]:
```suggestion
BendpointMovePreview.$inject = [
```

**Comment by philippfromme** [comment]
@barmac Is previewing while moving shapes not supposed to work?

**Comment by barmac** [comment]
Ready again.

File: lib/features/bendpoints/BendpointSnapping.js (hunk 2 above)
Comment (philippfromme) [reply lib/features/bendpoints/BendpointSnapping.js]: I don't think we should deactivate bendpoint snapping. Instead we should use `setSnapped` to mark the event as snapped if bendpoint snapping decides to snap it so it won't be snapped by grid snapping again.

If we decativate bendpoint snapping you can't snap to existing bendpoints with higher priority if they are not on the grid:

Comment (barmac) [reply lib/features/bendpoints/BendpointSnapping.js]: Cool. I'm reverting the commit about separation of features.

**Comment by barmac** [comment]
@philippfromme feel free to merge this

**Pull request merged**